flybywiresim/a32nx
# fix(nd): prettier ND with more formatting

## Summary of Changes


- [x] Separated degree symbol from waypointTrack variable
- [x] Aligned the value and symbol for waypointTrack
- [x] Created a new class for DegreeSymbol
- [x] Separate distance display logic (greater than and less than 20)
- [x] Adjusted values to smoothen transition for VORDME distance info on ND
- [x] ND Chronometer background
- [x] ROSE VOR and LS Page fixes with proper formatting and color
- [x] PPOS (Present position, when waypoint not loaded and C/D)
- [x] Black masking behind left VOR indicator to prevent drawn flight route from showing through

NOTE: 
1. ADF section wasn't modified individually as the variables are tied with VOR.

2. ~~On IRL displays, decimals have a smaller size on both Waypoint Distance indicator (top right) and VOR DME distance (bottom), it is left untouched for now.~~ Added decimal split and resizing wherever applicable, credits to @tracernz 🎉 

3. PPOS has not been implemented true to logic. It will only appear when no waypoint is loaded for now. Will implement logic with flightphase changes in a different PR.

## Screenshots (if necessary)



### BEFORE:
![image](https://user-images.githubusercontent.com/29005005/110492329-16ffe780-8118-11eb-94e6-6ae6694b4a33.png)
![image](https://user-images.githubusercontent.com/29005005/110492556-4f9fc100-8118-11eb-9a24-1c36bd5ee64b.png)

### AFTER:
![image](https://user-images.githubusercontent.com/29005005/110492388-23844000-8118-11eb-9d65-1bdd9337e0ae.png)
![image](https://user-images.githubusercontent.com/29005005/110492424-2e3ed500-8118-11eb-8fe4-445708beac0c.png)
![image](https://user-images.githubusercontent.com/29005005/110492489-3f87e180-8118-11eb-9f53-4187d7d0f8f8.png)
![image](https://user-images.githubusercontent.com/29005005/110492525-4878b300-8118-11eb-8ca2-cd1917484845.png)
![image](https://user-images.githubusercontent.com/29005005/111266152-215a3e00-8650-11eb-8123-e555085e15d1.png)
![image](https://user-images.githubusercontent.com/29005005/111266165-261ef200-8650-11eb-847f-b82cf75c7719.png)



## References


![image](https://user-images.githubusercontent.com/29005005/110492613-5deddd00-8118-11eb-8cdc-b82222ea37c1.png)
![image](https://user-images.githubusercontent.com/29005005/110492642-63e3be00-8118-11eb-8f23-4dab2b521a19.png)
![image](https://user-images.githubusercontent.com/29005005/110495513-942c5c00-811a-11eb-9510-90ec3081dac0.png)
![image](https://user-images.githubusercontent.com/29005005/110493009-8b3a8b00-8118-11eb-92ac-4c7957fda905.png)

https://youtu.be/TEi-WX08ov8?t=17
https://youtu.be/syr4G3pokgU?t=99




## Additional context



Discord username (if different from GitHub): sidnov#8337

## Testing instructions


1. Spawn C&D, check the updated ND
2. Align ADIRS, see the changes
3. Add VORs, waypoints, and fly through them.
4. Observe for any overlap in ARC, ROSE VOR and ILS distances/names/values on the top right of ND.
5. Observe distance decimal values appear only when Waypoint<20NM
6. Additionally, check VOR DME distance displayed on the bottom of the ND, updated with the same logic/value.
7. ND Chronometer has an improved background.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
